File: cifrado/escritorio/codigo/javascript/cifrado.js
   function buscarCaracterEnAlfabeto(c, alfabeto) {
            for (i = 0; i < alfabeto.length; i++) {
                if (alfabeto[i] == c)
                    return i;
            }
            return 0;
        }

 function cifrarLetra(c,alfabeto,desplazamiento) {
            var pos = buscarCaracterEnAlfabeto(c, alfabeto)+desplazamiento;
            if (pos >= alfabeto.length) 
                pos = pos - alfabeto.length;
                return alfabeto[pos];
        }

        function cifrarPalabra(result,palabra, alfabeto, desplazamiento) {
            for ( q = 0; q < palabra.length; q++) {
                result += cifrarLetra(palabra[q], alfabeto, desplazamiento);
            }
            return result;
        }

        function cifrarInverso(result,palabra){
             for(v=palabra.length-1;v>=0;v--){
                  result+=palabra[v];
             }
 			return result;
		}

		function concatenar2Cadenas(palabra1,palabra2){
			return palabra1+""+palabra2;
		}


		function cifrarGrupal(grupo,indice,resultado,palabra){
 			if(indice<palabra.length){

 			return cifrarGrupal(grupo,indice+grupo,cifrarInverso(resultado,palabra.substring(indice,indice+grupo)),palabra);
 			}

              return concatenar2Cadenas(resultado,palabra.substring(indice,palabra.length));

		}

File: cifrado/escritorio/codigo/c#/cifrado.cs
namespace WindowsFormsApplication3
{
    internal class cifrado
    {

        public int buscarCaracterEnAlfabeto(char c, string alfabeto)
        {
            for (int i = 0; i < alfabeto.Length; i++)
            {
                if (alfabeto[i] == c)
                    return i;
            }
            return 0;
        }


        //abcdefghijklmnñopqrstuvwxyz    27
        private char cifrarLetra(char c, string alfabeto, int desplazamiento)
        {
            int pos = buscarCaracterEnAlfabeto(c, alfabeto);
            if (pos > 0)
                pos = pos + desplazamiento;
            else
                return c;

            if (pos >= alfabeto.Length)
                pos = pos - alfabeto.Length;
               return alfabeto[pos];
        }

        public string cifrarPalabra(string result, string palabra, string alfabeto, int desplazamiento)
        {
            for (int i = 0; i < palabra.Length; i++)
            {    
                result += cifrarLetra(palabra[i], alfabeto, desplazamiento);
            }
            return result;
        }
    }
}